torch/csrc/autograd/python_function.cpp

```diff
@@ -507,6 +507,13 @@ static void _wrap_outputs(
     return results;
   };
 
+  auto view_as_fn = [](const at::Tensor& x, const at::Tensor& y) -> at::Tensor {
+    pybind11::gil_scoped_acquire gil;
+    py::object py_x = py::cast(x);
+    py::function py_view_as_method = py_x.attr("view_as");
+    return py_view_as_method(y).cast<at::Tensor>();
```

Comment:
Could you do a quick local check with %timeit to make sure this is not making this case significantly slower please?
If it does, we should use our direct APIs via THPVariable_wrap and cie to avoid the pybind overhead.

Reply:
I did a quick test, and it seems that this PR adds 0.3-4 us of overhead, but avoiding use of pybind can cut 0.1 us of that away.
- Current: 2.66 us
- Without pybind: 2.95 us
- With pybind 3.05 us

I've updated this PR to avoid using pybind to avoid that overhead.